airbyte-integrations/connectors/destination-selectdb/src/main/java/io/airbyte/integrations/destination/selectdb/SelectdbDestination.java

```diff
@@ -0,0 +1,170 @@
+      for (ConfiguredAirbyteStream stream : configuredCatalog.getStreams()) {
+
+        final DestinationSyncMode syncMode = stream.getDestinationSyncMode();
+        if (syncMode == null) {
+          throw new IllegalStateException("Undefined destination sync mode");
+        }
+
+        final String streamName = stream.getStream().getName();
+        final String tableName = namingResolver.getIdentifier(streamName);
+        final String tmpTableName = namingResolver.getTmpTableName(streamName);
+        final Path tmpPath = destinationDir.resolve(tmpTableName + ".csv");
+        if (conn == null)
+          checkSelectdbAndConnect(config);
+        Statement stmt = conn.createStatement();
+        stmt.execute(createTableQuery(tableName));
+        if (syncMode == DestinationSyncMode.OVERWRITE) {
+          stmt.execute(truncateTable(tableName));
+        }
+        CSVFormat csvFormat = CSVFormat.DEFAULT
+                .withSkipHeaderRecord()
+                .withDelimiter(SelectdbCopyInto.CSV_COLUMN_SEPARATOR)
+                .withQuote(null)
+                .withHeader(
+                        JavaBaseConstants.COLUMN_NAME_AB_ID,
+                        JavaBaseConstants.COLUMN_NAME_EMITTED_AT,
+                        JavaBaseConstants.COLUMN_NAME_DATA);
+        final FileWriter fileWriter = new FileWriter(tmpPath.toFile(), Charset.defaultCharset(), false);
+        final CSVPrinter printer = new CSVPrinter(fileWriter, csvFormat);
+        SelectdbCopyInto sci = new SelectdbCopyInto(
+                tmpPath,
+                SelectdbConnectionOptions.getSelectdbConnection(config, tableName),
+                new LabelInfo("",tableName),
+                http.getClient(),
+                JavaBaseConstants.COLUMN_NAME_AB_ID,
+                JavaBaseConstants.COLUMN_NAME_EMITTED_AT,
+                JavaBaseConstants.COLUMN_NAME_DATA);
+        writeConfigs.put(streamName, new SelectdbWriteConfig(sci, printer, csvFormat));
+      }
+    } catch (SQLException | ClassNotFoundException e) {
+      LOGGER.error("Exception while creating Selectdb destination table: ", e);
+      throw new SQLException(e);
+    } catch (IOException e) {
+      LOGGER.error("Exception while handling temporary csv files : ", e);
+      throw new IOException(e);
+    } finally {
+      if (conn != null)
+        conn.close();
+    }
+    return new SelectdbConsumer(writeConfigs, configuredCatalog, outputRecordCollector);
+  }
+
+
+  protected void checkSelectdbAndConnect(JsonNode config) throws ClassNotFoundException, SQLException {
+    SelectdbConnectionOptions selectdbConnection = SelectdbConnectionOptions.getSelectdbConnection(config, "");
+    String dbUrl = String.format(DB_URL_PATTERN, selectdbConnection.getJdbcUrl(), selectdbConnection.getDb());
+    Class.forName(JDBC_DRIVER);
+    conn = DriverManager.getConnection(dbUrl, selectdbConnection.getUser(), selectdbConnection.getPwd());
+  }
+
+  protected String createTableQuery(String tableName) {
```

Comment:
consider moving Selectdb related queries/operations to a separate class SelectdbOperations. Instead of having Selectdb connections, queries, urls  scattered around classes they should be encapsulated behind a class that is used for performing operations on the database

Reply:
done